lib/Onnxifi/Base.h

```diff
@@ -71,71 +56,39 @@ class BackendId {
   /// \returns the whether use onnx or not.
   bool getUseOnnx() const { return useOnnx_; }
 
-  /// \returns the whether use HostManager for inference or not.
-  bool getUseHostManager() const { return useHostManager_; }
-
   /// \returns HostManager associated with the BackendId.
-  HostManager &getHostManager() { return hostManager_; }
+  runtime::HostManager &getHostManager() { return *hostManager_; }
 
-  /// \returns the backend id.
-  int getID() const { return id_; }
-
-  /// \returns concurrency for the backend.
-  int getConcurrency() const { return concurrency_; }
-
-  /// \returns the glow Backend of this BackendId.
-  glow::Backend *getGlowBackend() { return glowBackend_.get(); }
-
-  /// Run the network named by \p networkName using HostManager with context \p
-  /// ctx afterwhich the result callback \p cb will be called.
-  void runOnHostManager(llvm::StringRef networkName,
-                        std::unique_ptr<Context> ctx, ResultCBTy cb) {
-    // TODO enable once HostManager is landed.
-    // hostManager_->runNetwork(networkName, std::move(ctx), std::move(cb));
-  }
+  // \returns a unique_ptr to a new HostManager for the given BackendKind \p
+  // kind.
+  static std::unique_ptr<runtime::HostManager>
+  createHostManager(glow::BackendKind kind);
 
 private:
-  int id_;
+  runtime::HostManager *hostManager_;
```

Comment:
Does this need to be a raw pointer?

Reply:
We could use a reference, would you prefer that? It can't be a unique_ptr due to lack of ownership and I don't think a shared_ptr is necessary.

Reply:
I'm good regarding pointer/reference. Just wanted to make sure it's not possible that the pointer is invalidated out from under you.

Reply:
Right now, no, HostManager effectively live forever until the GlowOnnxifiManager singleton dies (which could be improved but will require some reference counting and stuff)